inc/Config/DataSource/HttpDataSource.php

```diff
@@ -63,6 +65,31 @@ public static function preprocess_config( array $config ): array|WP_Error {
 		);
 	}
 
+	/**
+	 * Performs a migration if the config is out of date.
+	 */
+	private static function perform_migration( array $config ): array {
+		// By default, we want to have an active data source.
+		if ( ! isset( $config['active'] ) ) {
+			$config['active'] = true;
+		}
+
+		if ( static::SERVICE_SCHEMA_VERSION === $config['service_config']['__version'] ) {
+			return $config;
+		}
+
+		$migrated_config = static::migrate_config( $config );
+
+		if ( is_wp_error( $migrated_config ) ) {
+			$config['active'] = false;
+		} else {
+			$config['service_config']['__version'] = static::SERVICE_SCHEMA_VERSION;
```

Comment:
This should probably be delegated to the `migrate_config` version. Maybe there are scenarios where they don't want to advance to the latest version because there is missing info. I think we want to completely delegate to `migrate_config` and remove this method entirely.

Reply:
That's a fair point. Having us control would negate the point of letting data source owners take charge of the migration process.